client/src/test/scala/org/apache/celeborn/client/commit/SkewHandlingWithoutMapRangeValidatorTest.scala

```diff
@@ -0,0 +1,257 @@
+
+class SkewHandlingWithoutMapRangeValidatorTest extends CelebornFunSuite {
+
+  private var validator: SkewHandlingWithoutMapRangeValidator = _
+
+  test("testProcessSubPartitionFirstTime") {
+    // Setup
+    validator = new SkewHandlingWithoutMapRangeValidator
+    val partitionId = 1
+    val startMapIndex = 10
+    val endMapIndex = 0
+    val metadata = new CommitMetadata
+    val expectedMapperCount = 20
+
+    // Execute
+    val (success, message) = validator.processSubPartition(
+      partitionId,
+      startMapIndex,
+      endMapIndex,
+      metadata,
+      expectedMapperCount)
+
+    success shouldBe true
+    message shouldBe ""
+    validator.isPartitionComplete(partitionId) shouldBe false
+
+    // Verify current metadata
+    val currentMetadata = validator.currentCommitMetadata(partitionId)
+    currentMetadata shouldNot be(null)
+    currentMetadata shouldBe metadata
+  }
+
+  test("testProcessMultipleSubPartitions") {
+    // Setup
+    validator = new SkewHandlingWithoutMapRangeValidator
+    val partitionId = 1
+    val startMapIndex = 10
+    val metadata1 = new CommitMetadata(5, 1000)
+    val metadata2 = new CommitMetadata(3, 500)
+
+    // Process first sub-partition
+    validator.processSubPartition(partitionId, startMapIndex, 0, metadata1, startMapIndex)
+
+    // Process second sub-partition
+    val (success, message) = validator.processSubPartition(
+      partitionId,
+      startMapIndex,
+      1,
+      metadata2,
+      startMapIndex)
+
+    // Verify
+    success shouldBe true
+    message shouldBe ""
+    validator.isPartitionComplete(partitionId) shouldBe false
+
+    // Verify current metadata
+    val currentMetadata = validator.currentCommitMetadata(partitionId)
+    currentMetadata shouldNot be(null)
+    metadata1.addCommitData(metadata2)
+    currentMetadata shouldBe metadata1
+
+  }
+
+  test("testPartitionComplete") {
+    // Setup - processing all sub-partitions from 0 to 9 for a total of 10
+    validator = new SkewHandlingWithoutMapRangeValidator
+    val partitionId = 1
+    val startMapIndex = 10
+
+    // Process all sub-partitions
+    for (i <- 0 until startMapIndex) {
+      val metadata = new CommitMetadata()
+      validator.processSubPartition(partitionId, startMapIndex, i, metadata, startMapIndex)
+    }
+
+    validator.isPartitionComplete(partitionId) shouldBe true
+  }
+
+  test("testInvalidStartEndMapIndex") {
+    // Setup - invalid case where startMapIndex <= endMapIndex
+    validator = new SkewHandlingWithoutMapRangeValidator
+    val partitionId = 1
+    val startMapIndex = 5
+    val endMapIndex = 5 // Equal, which should fail
+    val metadata = new CommitMetadata()
+
+    // This should throw IllegalArgumentException
+    try {
+      validator.processSubPartition(partitionId, startMapIndex, endMapIndex, metadata, 20)
+    } catch {
+      case e: IllegalArgumentException => e.isInstanceOf[IllegalArgumentException] shouldBe true
```

Comment:
Consider using `intercept[IllegalStateException]` instead of try-catch for more concise and precise exception testing.